api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -122,7 +123,7 @@ public User getUserByEmail(String email) {
 	@Override
 	@SuppressWarnings("unchecked")
 	public User getUserByActivationKey(String activationKey) {
-		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.eq("user_activation_key", activationKey)).uniqueResult();	
+		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.ilike("activation_key", activationKey, MatchMode.START)).uniqueResult();	
```

Comment:
I think for this to be more correct you need to actually compare the keys after you have stripped off the ttl